### ISSUE TITLE
Only open Intercom chat once after opening push notification

### ISSUE THREAD

**Issue by ntoskrnl** [intercom/intercom-android#105]
In `onCreate()` of my app's main activity I call
`Global.intercomClient().openFromPush(getIntent())`

Please consider the following test case:
1.  Open push notification from intercom, the app starts and I see Intercom chat activity as expected.
2.  Tap back 2 times to close the chat window and the app. 
3. Go back to the application from *recent apps*.

I should return to my app, but instead I see the chat again (because Intent is the same), so I have to tap BACK to return to the app.
Is there a way to prevent this behaviour?

**Comment by harkin** [comment]
That's an interesting issue. I don't want to leave this hanging for too long without a response. We're thinking and talking about this. 

First reaction is that you're passing the same Uri in both times so it's reasonable for Intercom to react the same both times. This puts the burden of managing when to call `intercom.openGcmMessage()` on the host app and that's not something we _really_ want to do.

I'll update this tomorrow with a more well thought out opinion

**Comment by ntoskrnl** [comment]
Thank you for your quick reply. This is a very minor issue, just that our testers find this behavior a bit odd...

Indeed, the intent (and thus the uri) is the same. I was thinking how to tell the activity that this intent has already been handled... There should be some boolean flag that would mae activity forget about the first intent (true if intercom has been shown for this uri), but where should I save this flag? 

Looking forward to hearing your opinion :)

**Comment by DaleCantwell** [comment]
@ntoskrnl there are a few options to fix this issue. In the long term we will look at modifying the method to pass the intent on with an additional extra 'wasUsedByIntercomSDK' field to allow host apps to know we have used this intent. For now there are a few host solutions:

This is essentially what we will do inside the SDK in the next release
```JAVA
        Intent intent = getIntent();
        boolean isUsed = intent.getExtras() != null && intent.getExtras().getBoolean("isUsed");
        if(!isUsed && Intercom.client().openGCMMessage(intent.getData())) {
            intent.putExtra("isUsed", true);
        }
```

You could check if the intent has been launched before
```JAVA
        Intent intent = getIntent();
        boolean launchedFromHistory = intent != null && (intent.getFlags() & Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY) != 0;
        if(!launchedFromHistory) {
             Intercom.client().openGCMMessage(getIntent().getData());
        }
```
You could totally clear the intent
```JAVA
        if(Intercom.client().openGCMMessage(getIntent().getData())) {
            onNewIntent(new Intent());
        }
```
Let me know if that works for you.

**Comment by DaleCantwell** [comment]
Fixed in v1.1.5

**Comment by ntoskrnl** [comment]
Thank you! We will try! :+1:
